lib/event.c

```diff
@@ -952,6 +1265,95 @@ void _event_add_read_write(const struct xref_eventsched *xref,
 			   struct event_loop *m, void (*func)(struct event *),
 			   void *arg, int fd, struct event **t_ptr)
 {
+#if EPOLL_ENABLED
+	struct frr_epoll_event *new_frr_ev;
+	struct epoll_event *new_ev;
+	bool has_new;
+	int dir = xref->event_type;
+	struct event *thread = NULL;
+	struct event **thread_array;
+
+	if (dir == EVENT_READ)
+		frrtrace(9, frr_libfrr, schedule_read, m, xref->funcname,
+			 xref->xref.file, xref->xref.line, t_ptr, fd, 0, arg,
+			 0);
+	else
+		frrtrace(9, frr_libfrr, schedule_write, m, xref->funcname,
+			 xref->xref.file, xref->xref.line, t_ptr, fd, 0, arg,
+			 0);
+
+	assert(fd >= 0);
+	if (fd >= m->fd_limit)
+		assert(!"Number of FD's open is greater than FRR currently configured to handle, aborting");
+
+	frr_with_mutex (&m->mtx) {
+		if (t_ptr && *t_ptr)
+			/* thread is already scheduled; don't reschedule */
+			break;
+
+		/*
+		 * default: a new frr_epoll_event for the new request.
+		 * Queue the new event at the end of .new_event_ptrs array
+		 */
+		nfds_t queuepos = m->handler.new_event_count;
+
+		if (dir == EVENT_READ)
+			thread_array = m->read;
+		else
+			thread_array = m->write;
+
+		new_frr_ev = m->handler.new_events + fd;
+		has_new = (new_frr_ev->index >= 0);
+
+		/*
+		 * if we already have a new_event for our file descriptor, find
+		 * and use it
+		 */
+		if (has_new) {
+			queuepos = new_frr_ev->index;
+#ifdef DEV_BUILD
+			zlog_debug("event_add_read_write: DEV_BUILD");
```

Comment:
I would not add a zlog_debug for DEV_BUILD compile option.  This will clutter up log files too much

Reply:
This line is deleted.